ui/org.eclipse.pde.launching/src/org/eclipse/pde/launching/AbstractPDELaunchConfiguration.java

```diff
@@ -189,6 +192,29 @@ private String[] updateVMArgumentWithAdditionalArguments(String[] args, boolean
 		return args;
 	}
 
+
+	private boolean isEclipseBundleGreaterThanVersion(int major, int minor) {
+		PDEState pdeState = TargetPlatformHelper.getPDEState();
+		if (pdeState != null) {
+			try {
+				Optional<IPluginModelBase> platformBaseModel = Arrays.asList(pdeState.getTargetModels()).stream().filter(x -> Objects.nonNull(x.getBundleDescription())).filter(x -> ("org.eclipse.platform").equals(x.getBundleDescription().getSymbolicName()))//$NON-NLS-1$
```

Comment:
In reply to the conversation at the same place in the previous PR:

> ok, we will still need one not null filter(x -> Objects.nonNull(x.getBundleDescription()) before running through x.getBundleDescription().getSymbolicName().

Yes. Furthermore you could use `Arrays.stream()` instead of `Arrays.asList().stream()`.

> I also prefer to go the normal for each loop way for read and debug but this might be faster I guess and if we provide normal for loop codes, we get suggestions to use new ways to attract contributions.

As somebody who contributes for not very long I think that always using the newest thing just because it is newer does not attract more contributors.
IMHO clean code that uses the 'best' (fastest and especially most readable) style attracts more contributors. Sometimes this is a stream, sometimes a classic for-loop and sometimes a nested block should be refactored into a separate method.
But what's readable is of course also a matter of preference. :)

Besides that: Just checking all target-models is not the 100% right thing to do. Actually you should check the list of plug-ins that participate in the launched application since somebody could also have "org.eclipse.platform" in its workspace and if "org.eclipse.platform" is not a singleton there could be multiple versions in the active TP.

Unfortunately only the sub-classes have a List/Map/Set of actually launched plug-ins. So an abstract method to get the List<IPluginModelBase> of launched o.e.platform plug-ins has to be added and that list can be checked (e.g. if all plug-ins are above version `4.24`.

The question is if we want to make this method abstract and protected so that all sub-classes of `AbstractPDELaunchConfiguration` have to implement it or if we not make it abstract and only package and return a default value (that always results either in a true or false result?) only `EquinoxLaunchConfiguration` and `EclipseApplicationLaunchConfiguration` can overwrite it with the really launched plug-ins.

Reply:
Current change is a good workaround for the problem the user is facing,  We can discuss further in a new issue and take the best approach.
@vik-chand WDYT ?

Reply:
Created the followup issue https://github.com/eclipse-pde/eclipse.pde/issues/105